tt_metal/programming_examples/sfpu_eltwise_chain/kernels/dataflow/reader.cpp

```diff
@@ -0,0 +1,65 @@
+// SPDX-FileCopyrightText: © 2025 Tenstorrent AI ULC
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "dataflow_api.h"
+#include "tt-metalium/constants.hpp"
+
+#include <cstdint>
+#include <cstring>
+
+/**
+ * @brief Converts a 32-bit IEEE 754 float to 16-bit bfloat16 format.
+ *
+ * This function performs a simple truncation conversion from float32 to bfloat16
+ * by extracting the upper 16 bits (sign, exponent, and upper 7 bits of mantissa)
+ * of the IEEE 754 float representation. The lower 16 bits of the mantissa are
+ * discarded, which may result in precision loss but maintains the same range
+ * as float32.
+ *
+ * @param value The input 32-bit floating point value to convert
+ * @return uint16_t The resulting 16-bit bfloat16 value in its binary representation
+ *
+ * @note This implementation uses simple truncation without rounding, which may
+ *       introduce quantization errors for values that cannot be exactly
+ *       represented in bfloat16 format - it is sufficient for this example.
+ */
+inline uint16_t float_to_bfloat16(float value) {
+    uint32_t tmp;
+    std::memcpy(&tmp, &value, sizeof(tmp));
+    return static_cast<uint16_t>(tmp >> 16);
+}
```

Comment:
Why are we doing cast in reader? This should be handled by the unpacker.

1. Create CB in FP32 format
2. Put tile in CB
3. If `dst_fp32_acc_en` is false in kernel config. It automatically gets unpacked into BFP16 in DST (maybe we should mention this fact in the tech report?)

Reply:
Ah I see. Then let's keep this here. Make sense.